integrations/ragas/src/haystack_integrations/components/evaluators/ragas/metrics.py

```diff
@@ -0,0 +1,335 @@
+            raise ValueError(msg)
+
+    @staticmethod
+    def extract_results(
+        output: Result, metric: RagasMetric, metric_params: Optional[Dict[str, Any]]
+    ) -> List[MetricResult]:
+        try:
+            metric_name = ""
+            if metric == RagasMetric.ASPECT_CRITIQUE and metric_params:
+                if "name" in metric_params:
+                    metric_name = metric_params["name"]
+                elif "aspect" in metric_params:
+                    metric_name = metric_params["aspect"].value
+            else:
+                metric_name = metric.value
+            output_scores: List[Dict[str, float]] = output.scores.to_list()
+            return [MetricResult(name=metric_name, score=metric_dict[metric_name]) for metric_dict in output_scores]
+        except KeyError as e:
+            msg = f"Ragas evaluator did not return an expected output for metric '{e.args[0]}'"
+            raise ValueError(msg) from e
+
+
+METRIC_DESCRIPTORS = {
+    RagasMetric.ANSWER_CORRECTNESS: MetricDescriptor.new(
+        RagasMetric.ANSWER_CORRECTNESS, answer_correctness, InputConverters.question_response_ground_truth  # type: ignore
+    ),
+    RagasMetric.FAITHFULNESS: MetricDescriptor.new(
+        RagasMetric.FAITHFULNESS, faithfulness, InputConverters.question_context_response  # type: ignore
+    ),
+    RagasMetric.ANSWER_SIMILARITY: MetricDescriptor.new(
+        RagasMetric.ANSWER_SIMILARITY, answer_similarity, InputConverters.response_ground_truth  # type: ignore
+    ),
+    RagasMetric.CONTEXT_PRECISION: MetricDescriptor.new(
+        RagasMetric.CONTEXT_PRECISION, context_precision, InputConverters.question_context_ground_truth  # type: ignore
+    ),
+    RagasMetric.CONTEXT_UTILIZATION: MetricDescriptor.new(
+        RagasMetric.CONTEXT_UTILIZATION,
+        context_utilization,
+        InputConverters.question_context_response,  # type: ignore
+    ),
+    RagasMetric.CONTEXT_RECALL: MetricDescriptor.new(
+        RagasMetric.CONTEXT_RECALL, context_recall, InputConverters.question_context_ground_truth  # type: ignore
+    ),
+    RagasMetric.ASPECT_CRITIQUE: MetricDescriptor.new(
+        RagasMetric.ASPECT_CRITIQUE, AspectCritique, InputConverters.question_context_response  # type: ignore
+    ),
+    RagasMetric.CONTEXT_RELEVANCY: MetricDescriptor.new(
+        RagasMetric.CONTEXT_RELEVANCY, context_relevancy, InputConverters.question_context  # type: ignore
+    ),
+    RagasMetric.ANSWER_RELEVANCY: MetricDescriptor.new(
+        RagasMetric.ANSWER_RELEVANCY, answer_relevancy, InputConverters.question_context_response  # type: ignore
+    ),
+}
+
+METRIC_ASPECTS = {
+    RagasMetricAspect.HARMFULNESS: harmfulness,
+    RagasMetricAspect.MALICIOUSNESS: maliciousness,
+    RagasMetricAspect.COHERENCE: coherence,
+    RagasMetricAspect.CORRECTNESS: correctness,
+    RagasMetricAspect.CONCISENESS: conciseness,
+}
```

Comment:
I'm not sure we want directly expose these pre-defined aspects as-is. The `AspectCritique` is the underlying metric class, so that should be the one we expose. 

If we get feedback that it'd also make sense to have these pre-defined ones, we can provide some pre-defined init params for the metric.

Reply:
We agreed to leave out all pre-defined metrics and consequently this is resolved.